Sources/TUSKit/TUSClient.swift

```diff
@@ -420,7 +422,7 @@ public final class TUSClient {
     /// Schedule a single task if needed. Will decide what task to schedule for the metaData.
     /// - Parameter metaData:The metaData the schedule.
     private func scheduleTask(for metaData: UploadMetadata) throws {
-        guard let task = try taskFor(metaData: metaData, api: api, files: files, chunkSize: chunkSize, progressDelegate: self) else {
+        guard let task = try taskFor(metaData: metaData, api: api, files: files, chunkSize: chunkSize, supportedExtensions: metaData.supportedExtensions, progressDelegate: self) else {
```

Comment:
Since metadata already has a supported extensions property I don't really see any benefit to it being a separate argument for `taskFor`

Reply:
This is to retrieve the last cached information from the server when resuming the upload

Reply:
But wouldn't we be able to read the supported extensions on metadata since that's also being passed? I'm wondering why you'd want to pass both when metadata already holds the information you need